File: M3Tools/Dialogs/Google/PermissionsDialog.vb
Imports System.Windows.Forms
Imports Google.Apis.Drive.v3.Data

Public Class PermissionsDialog
	Public Property Permission As New Permission

	Private ReadOnly __roles As String() = {"Owner", "Organizer", "File Organizer", "Writer", "Commenter", "Reader"}
	Private ReadOnly __types As String() = {"User", "Group", "Domain", "Anyone"}

	Private Sub SetPermission(sender As Object, e As EventArgs) Handles OK_Button.Click
		Me.DialogResult = DialogResult.OK
		Me.Close()
	End Sub

	Private Sub Cancel(sender As Object, e As EventArgs) Handles Cancel_Button.Click
		Me.DialogResult = DialogResult.Cancel
		Me.Close()
	End Sub

	Private Sub PermissionsDialogLoaded(sender As Object, e As EventArgs) Handles MyBase.Load
		bsRoles.DataSource = __roles
		bsTypes.DataSource = __types
		cbx_Role.SelectedItem = "Reader"
		cbx_Type.SelectedItem = "Anyone"
	End Sub

	Private Sub RoleChanged(sender As Object, e As EventArgs) Handles cbx_Role.SelectedIndexChanged
		Permission.Role = CStr(cbx_Role.SelectedItem).Replace(" "c, "")
	End Sub

	Private Sub TypeChanged(sender As Object, e As EventArgs) Handles cbx_Type.SelectedIndexChanged
		Permission.Type = CStr(cbx_Type.SelectedItem).Replace(" "c, "")
	End Sub
End Class

File: M3Tools/Dialogs/Google/PermissionsDialog.cs
using System;
using System.Windows.Forms;
using Google.Apis.Drive.v3.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace SPPBC.M3Tools
{

    public partial class PermissionsDialog
    {
		/// <summary>
		/// The set of permissions
		/// </summary>
        public Permission Permission { get; set; } = new Permission();

        private readonly string[] __roles = new[] { "Owner", "Organizer", "File Organizer", "Writer", "Commenter", "Reader" };
        private readonly string[] __types = new[] { "User", "Group", "Domain", "Anyone" };

		/// <summary>
		/// 
		/// </summary>
        public PermissionsDialog()
        {
            InitializeComponent();
        }

        private void SetPermission(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
            Close();
        }

        private void Cancel(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void PermissionsDialogLoaded(object sender, EventArgs e)
        {
            bsRoles.DataSource = __roles;
            bsTypes.DataSource = __types;
            cbx_Role.SelectedItem = "Reader";
            cbx_Type.SelectedItem = "Anyone";
        }

        private void RoleChanged(object sender, EventArgs e)
        {
            Permission.Role = Conversions.ToString(cbx_Role.SelectedItem).Replace(" ", "");
        }

        private void TypeChanged(object sender, EventArgs e)
        {
            Permission.Type = Conversions.ToString(cbx_Type.SelectedItem).Replace(" ", "");
        }
    }
}
